tiledb/sm/query/readers/sparse_global_order_reader.cc

```diff
@@ -1996,9 +1921,196 @@ void SparseGlobalOrderReader<BitmapType>::process_slabs(
+  OffType offset_div =
+      elements_mode_ ? datatype_size(array_schema_.type(name)) : 1;
+  if (name == constants::timestamps) {
+    copy_timestamps_tiles(
+        num_range_threads, result_cell_slabs, cell_offsets, query_buffer);
+  } else if (name == constants::delete_condition_index) {
+    // Copy fixed size data.
+    copy_delete_meta_tiles(
+        num_range_threads, result_cell_slabs, cell_offsets, query_buffer);
+  } else if (var_sized) {
+    copy_offsets_tiles<OffType>(
+        name,
+        num_range_threads,
+        nullable,
+        offset_div,
+        result_cell_slabs,
+        cell_offsets,
+        query_buffer,
+        var_data);
+  } else {
+    copy_fixed_data_tiles(
+        name,
+        num_range_threads,
+        is_dim,
+        nullable,
+        dim_idx,
+        cell_size,
+        result_cell_slabs,
+        cell_offsets,
+        query_buffer);
+  }
+
+  uint64_t var_buffer_size = 0;
+  if (var_sized) {
+    // Adjust the offsets buffer and make sure all data fits.
+    auto&& [caused_overflow, var_buff_size] = compute_var_size_offsets<OffType>(
+        stats_, result_cell_slabs, cell_offsets, query_buffer);
+    user_buffers_full |= caused_overflow;
+    var_buffer_size = var_buff_size;
+
+    // Save the last field to overflow.
+    if (caused_overflow) {
+      last_field_to_overflow = name;
+    }
+
+    // Now copy the var size data.
+    copy_var_data_tiles(
+        num_range_threads,
+        offset_div,
+        var_buffer_size,
+        result_cell_slabs,
+        cell_offsets,
+        query_buffer,
+        var_data);
+  }
+
+  // Adjust buffer sizes.
+  auto total_cells = cell_offsets[result_cell_slabs.size()];
+  if (var_sized) {
+    *query_buffer.buffer_size_ = total_cells * sizeof(OffType);
+
+    if (offsets_extra_element_)
+      (*query_buffer.buffer_size_) += sizeof(OffType);
+
+    *query_buffer.buffer_var_size_ = var_buffer_size * offset_div;
+  } else {
+    *query_buffer.buffer_size_ = total_cells * cell_size;
+  }
+
+  if (nullable) {
+    *buffers_[name].validity_vector_.buffer_size() = total_cells;
+  }
+
+  // For delete timestamps, since they get processed at the same time as
+  // the delete condition indexes, we need to adjust the buffer size.
+  if (name == constants::delete_condition_index) {
+    *buffers_[constants::delete_timestamps].buffer_size_ =
+        total_cells * constants::timestamp_size;
+  }
+
+  return user_buffers_full;
+}
+
+template <class BitmapType>
+void SparseGlobalOrderReader<BitmapType>::process_aggregates(
+    const uint64_t num_range_threads,
+    const std::string name,
+    std::vector<uint64_t>& cell_offsets,
+    std::vector<ResultCellSlab>& result_cell_slabs) {
+  auto& aggregates = aggregates_[name];
+
+  bool var_sized = false;
+  bool nullable = false;
+
+  if (name != constants::all_attributes) {
+    var_sized = array_schema_.var_size(name);
+    nullable = array_schema_.is_nullable(name);
+  }
```

Comment:
It just occurred to me reading here that all_attributes means we get some weird issues around some attributes being nullable or var sized when others aren't? No idea what the ramifications of that might mean.

Reply:
all_attributes might be a bad name? This is meant to represent * in SELECT COUNT(*) WHERE x > 10.